include/DREAM/Equations/Scalar/ConstantSPIVelocityPositionTerm.hpp

```diff
@@ -46,12 +50,29 @@ namespace DREAM {
                 this->initial_data=xp_init;
             }
         }
+        void SetTimeDelay(const real_t *t_delay, bool copy=true){
+            if(copy){
+                real_t *data_tmp=new real_t[nShard];
```

Comment:
I cannot find any corresponding ``delete`` for this one. And I hope that the input ``t_delay`` is deleted by the caller?